pkg/controllers/selection/preferences.go

```diff
@@ -101,7 +109,23 @@ func (p *Preferences) removeRequiredNodeAffinityTerm(pod *v1.Pod) *string {
 	// Remove the first term if there's more than one (terms are an OR semantic), Unlike preferred affinity, we cannot remove all terms
 	if len(terms) > 1 {
 		pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = terms[1:]
-		return ptr.String(fmt.Sprintf("spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution[0]=%s", pretty.Concise(terms[0])))
+		return ptr.String(fmt.Sprintf("removing: spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution[0]=%s", pretty.Concise(terms[0])))
 	}
 	return nil
 }
+
+func (p *Preferences) toleratePreferNoScheduleTaints(pod *v1.Pod) *string {
```

Comment:
Thoughts on making these all return error? `error` is more or less a nillable string, and I think it expresses the semantic more cleanly.

Reply:
Thinking about this further, are you suggesting that we return `error` instead of `*string`?  Or in addition to `*string`?

Reply:
Instead of *string.